plugin-server/functional_tests/ingestion.test.ts

```diff
@@ -317,108 +272,197 @@ test.concurrent('consumer updates timestamp exported to prometheus', async () =>
-            distinct_id: personIdentifier,
-        })
-        await waitForExpect(async () => {
-            const persons = await fetchPersons(clickHouseClient, teamId)
-            expect(persons).toContainEqual(
+                }),
                 expect.objectContaining({
-                    properties: expect.objectContaining({ $creator_event_uuid: initialIdentifyEventId }),
-                })
-            )
-        }, 10000)
-
-        // Then we create another event with the initial anonymous distinct id,
-        // shortly followed by another identify event but this time with the
-        // initial anonymous distinct id
-        const uuidOfEventThatShouldBeIdentified = new UUIDT().toString()
-        await capture(producer, teamId, initialDistinctId, uuidOfEventThatShouldBeIdentified, 'custom event')
-
-        const uuidOfIdentifyEvent = new UUIDT().toString()
-        await capture(producer, teamId, personIdentifier, uuidOfIdentifyEvent, '$identify', {
-            distinct_id: personIdentifier,
-            $anon_distinct_id: initialDistinctId,
-        })
+                    properties: expect.objectContaining({ $creator_event_uuid: secondEventId }),
+                }),
+            ])
+        )
+    }, 10000)
+
+    // Then we identify both ids
+    const uuidOfFirstIdentifyEvent = new UUIDT().toString()
+    await capture(producer, teamId, personIdentifier, uuidOfFirstIdentifyEvent, '$identify', {
+        distinct_id: personIdentifier,
+        $anon_distinct_id: initialDistinctId,
+    })
+    const uuidOfSecondIdentifyEvent = new UUIDT().toString()
+    await capture(producer, teamId, personIdentifier, uuidOfSecondIdentifyEvent, '$identify', {
+        distinct_id: personIdentifier,
+        $anon_distinct_id: secondEventId,
+    })
+
+    let events: ClickHouseEvent[] = []
+    await waitForExpect(async () => {
+        events = await fetchEvents(clickHouseClient, teamId)
+        expect(events.length).toBe(4)
+    }, 10000)
+
+    events.forEach((event) => {
+        expect(event?.person_id).toBeDefined()
+    })
 
-        await waitForExpect(async () => {
-            const [anonymousEvent] = await fetchEvents(clickHouseClient, teamId, uuidOfEventThatShouldBeIdentified)
-            const [identifyEvent] = await fetchEvents(clickHouseClient, teamId, uuidOfIdentifyEvent)
-            expect(anonymousEvent?.person_id).toBeDefined()
-            expect(identifyEvent?.person_id).toBeDefined()
-            expect(anonymousEvent.person_id).toEqual(identifyEvent.person_id)
-        }, 10000)
+    if (testIfPoEEmbraceJoinEnabled) {
+        // TODO: check person overrides table to have 1 entry
+        // TODO: event.person_id or override[event.person_id] is the same id for all of the events
     }
-)
+})
 
-testIfDelayEnabled(`events reference same person_id if two people merged shortly after`, async () => {
+test.concurrent(`chained merge results in an both linking to final person in person overrides`, async () => {
```

Comment:
Likewise for the previous test, can we test for user functionality rather than implementation detail?